asapdiscovery-genetics/asapdiscovery/genetics/blast.py

```diff
@@ -0,0 +1,264 @@
+    result_handle = NCBIWWW.qblast(
+        program, database, sequence, hitlist_size=nhits, alignments=alignments
+    )
+
+    # Create folder if doesn't already exists
+    Path(save_folder).mkdir(parents=True, exist_ok=True)
+    save_file = f"{save_folder}/{xml_file}"
+
+    with open(save_file, "w") as file:
+        blast_results = result_handle.read()
+        file.write(blast_results)
+
+    matches_df = parse_blast(save_file, verbose)
+    if save_csv:
+        matches_df.to_csv(f"{save_folder}/{save_csv}", index=False)
+
+    return matches_df
+
+
+class PDB_record:
+    def __init__(
+        self, label: str, query_seq: str, description: str, chain: int
+    ) -> None:
+        self.label = label
+        self.description = description
+        self.seq = query_seq
+        self.chain = chain
+
+
+class PDBentry:
+    def __init__(self, input_seq: str, input_type: str) -> None:
+        self.input = input_seq
+        self.type = input_type
+
+    def retrieve_pdb(self, results_folder: str, min_id_match=99, ref_only=False):
+        """Retrieve the PDB record of a given sequence."""
+        from pathlib import Path
+
+        import prody
+
+        record_name = "pdb_blast.xml"
+
+        # Find pdb matching given sequence
+        matches_df = get_blast_seqs(
+            self.input,
+            results_folder,
+            input_type=self.type,
+            xml_file=record_name,
+            nalign=500,
+            database="pdb",
+            verbose=False,
+        )
+        print(f"Saving blast results in {results_folder}/{record_name}")
+
+        # Load original sequence names and descriptors for reference
+        pdb_file_record = []
+        for seq_record in SeqIO.parse(self.input, self.type):
+            seq_id = seq_record.id
+            seq_description = seq_record.description
+            seq_chain = int(seq_record.id.split("|")[1].split(".")[-1])
+            pdb_file_record.append(
+                PDB_record(seq_id, seq_record.seq, seq_description, seq_chain)
+            )
+
+        best_scores = []
+        for query in matches_df["query"].unique():
+            best_scores.append(
+                matches_df.loc[matches_df["query"] == query, "score"].max()
+            )
+
+        match_hits = self.parse_pdb_blast_results(matches_df, min_score=min_id_match)
+
+        for i, hits in enumerate(match_hits):
+            if len(hits) == 0:
+                print("The record doesn't have a PDB with high confidence")
+                if i == 0:
+                    raise ValueError(
+                        "The reference sequence MUST have an existing PDB entry. Maybe check the sequence?"
+                    )
+            else:
+                # Return the pdb entry with the max alignment and max resolution (if there are multiple matches)
+                best_pdb_record = self.choose_best_pdb_entry(hits, best_scores[i])
+                filename = prody.fetchPDB(best_pdb_record, folder=results_folder)
```

Comment:
Would you be able to check if this BioPython (which we already import) can replace the ProDy version? 
https://biopython.org/docs/1.75/api/Bio.PDB.PDBList.html#Bio.PDB.PDBList.PDBList.retrieve_pdb_file

That way we can eliminate ProDy dependency 

Would look something like:

```python
from Bio.PDB import PDBList
pdbl = PDBList()
fname = pdbl.retrieve_pdb_file(pdb_code)
```

Reply:
Unfortunately PDBList doesn't work. It says it can't find the PDB after I tried with a few pdb codes.